### PR TITLE
Initial commit to add conditional-display into builder

### PR DESCRIPTION
**Pull request by mariarivera30**
Conditional-display in builder

### CODE REVIEW

File: static/src/components/builder/components/ComponentRender/ComponentRender.vue [hunk 1]
```diff
@@ -85,6 +87,10 @@ export default {
           attrs: { id: 'test' },
           props: { checkboxList: this.form.checkboxList }
         };
+      } else if (this.selectedComponent === 'conditional-display-creator') {
+        return {
+          name: 'conditionaldisplay-creator'
```
Comment (markroseberry) [inline static/src/components/builder/components/ComponentRender/ComponentRender.vue]: Why did you hyphenate `conditional display creator` two different ways? It's confusing.

File: static/src/components/builder/store/modules/conditionalDisplay.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+const initialState = () => {
+  return {
+    id: utils.uniqueID(),
+    condition: '',
+    isValidForm: true
+  };
+};
+
+export const state = {
```
Comment (markroseberry) [inline static/src/components/builder/store/modules/conditionalDisplay.js]: You could do `state = initialState()`. No need to clone the return value.

File: static/src/components/builder/components/ConditionalDisplay/conditionalDisplayTemplate.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<conditional-display id="{{id}}" :condition="{{{condition}}}">
```
Comment (markroseberry) [inline static/src/components/builder/components/ConditionalDisplay/conditionalDisplayTemplate.html]: The `{{{ }}}` prevents html encoding, right? Will there be a problem if `condition` contains `"`?

File: static/src/components/builder/components/ConditionalDisplay/ConditionalDisplayForm.vue [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+}
+</style>
+<script>
+import * as types from '../../store/types';
+import { mapMutations, mapState } from 'vuex';
+export default {
+  name: 'CondditionalDisplayForm',
+  computed: {
+    ...mapState({
+      condition: state => state.conditionalDisplay['condition']
```
[the remark this answers is not in the document]
Comment (mariarivera30) [reply static/src/components/builder/components/ConditionalDisplay/ConditionalDisplayForm.vue]: its true python is getting on my way.

File: static/src/components/builder/components/ConditionalDisplay/conditionalDisplayTemplate.html (hunk 3 above)
Comment (mariarivera30) [reply static/src/components/builder/components/ConditionalDisplay/conditionalDisplayTemplate.html]: You are right. Im going to add a validation that checks if they use `"`,  to switch to `'`

File: static/src/components/builder/components/ComponentRender/ComponentRender.vue (hunk 1 above)
Comment (mariarivera30) [reply static/src/components/builder/components/ComponentRender/ComponentRender.vue]: I noticed this is not used will remove it.

File: static/src/components/builder/store/modules/conditionalDisplay.js [hunk 5]
```diff
@@ -9,16 +9,20 @@ const initialState = () => {
   };
 };
 
-export const state = {
-  ...initialState()
-};
+export const state = initialState();
 
+function* getErrors (state) {
+  if (state.condition.includes('"')) yield ['condition', ["Please avoid using [ \" ] and use [ ' ] instead."]];
```
Comment (markroseberry) [inline static/src/components/builder/store/modules/conditionalDisplay.js]: The error message shouldn't be inside an array. It should be:
`
['condition', "Please avoid using [ \" ] and use [ ' ] instead."]
`

`utils.toMultiDict` will append the error to an array for you.

**Pull request merged**